environment/commands/transfer-eth-erc20-to-near.js

```diff
@@ -96,89 +236,36 @@ class TransferETHERC20ToNear {
-                const ethClientContract = new Eth2NearClientContract(nearMasterAccount, clientAccount);
-                while (true) {
-                    // @ts-ignore
-                    const last_block_number = (await ethClientContract.last_block_number()).toNumber();
-                    const is_safe = await ethClientContract.block_hash_safe(blockNumber);
-                    if (!is_safe) {
-                        const delay = 10;
-                        console.log(`Eth2NearClient is currently at block ${last_block_number}. Waiting for block ${blockNumber} to be confirmed. Sleeping for ${delay} sec.`);
-                        await sleep(delay * 1000);
-                    } else {
-                        break;
-                    }
-                }
-
-                // @ts-ignore
-                const old_balance = await nearTokenContract.get_balance({
-                    owner_id: new_owner_id,
-                });
-                console.log(`Balance of ${new_owner_id} before the transfer is ${old_balance}`);
-
-                // @ts-ignore
-                await nearTokenContractBorsh.mint(
-                    proof_locker,
-                    new BN('300000000000000'),
-                    // We need to attach tokens because minting increases the contract state, by <600 bytes, which
-                    // requires an additional 0.06 NEAR to be deposited to the account for state staking.
-                    // Note technically 0.0537 NEAR should be enough, but we round it up to stay on the safe side.
-                    (new BN('100000000000000000000')).mul(new BN('600')),
-                );
-                console.log('Transferred');
-
-                // @ts-ignore
-                const new_balance = await nearTokenContract.get_balance({
-                    owner_id: new_owner_id,
-                });
-                console.log(`Balance of ${new_owner_id} after the transfer is ${new_balance}`);
+        const ethTokenLockerContract = new web3.eth.Contract(
+            // @ts-ignore
+            JSON.parse(fs.readFileSync(RainbowConfig.getParam('eth-locker-abi-path'))),
+            RainbowConfig.getParam('eth-locker-address'),
+        );
 
-                break;
-            }
-        }
+        const clientAccount = RainbowConfig.getParam('eth2near-client-account');
+        const ethClientContract = new Eth2NearClientContract(nearMasterAccount, clientAccount);
 
-        if (!logFound) {
-            console.log(`ERROR log not found for event ${lockedEvent}`);
+        if (transferLog.finished === undefined) {
+            await TransferETHERC20ToNear.approve({ ethERC20Contract, amount, ethSenderAccount });
+            transferLog = TransferETHERC20ToNear.loadTransferLog();
+        }
+        if (transferLog.finished === 'approve') {
+            await TransferETHERC20ToNear.lock({ ethTokenLockerContract, amount, nearReceiverAccount, ethSenderAccount });
```

Comment:
Is there a reason we are passing args as an object?

Reply:
Major benefit is no worry about sequence of arguments so it's less error prune when there're many args.